internal/pkg/template/templates/task/cf.yml

```diff
@@ -122,8 +122,17 @@ Resources:
                       "secretsmanager:ResourceTag/copilot-environment": !Ref Env
                   Resource:
                     - !Sub arn:${AWS::Partition}:secretsmanager:${AWS::Region}:${AWS::AccountId}:secret:*
-          - !Ref AWS::NoValue
-      {{- end}}
+          - PolicyName: 'PullSecrets'
+            PolicyDocument:
+              Version: '2012-10-17'
+              Statement:
+                - Effect: 'Allow'
+                  Action:
+                    - ssm:GetParameters
+                    - secretsmanager:GetSecretValue
+                  Resource: {{range $name, $valueFrom := .Secrets}}
```

Comment:
What about the secretsmanager secrets?

Reply:
.Secrets will have both secrets manager secrets and ssm params.

Reply:
oh yea my bad. I forgot that when `isARN` returns `false` it implies that it's an SSM parameter. 

Though I think ideally 💭 it'd be great if we can have separate policy statements like the `HasAppAndEnv`. Like one statement for `ssm` and one for `secretsmanager`. What do you think?

Reply:
I think current logic is pretty clean and less complex. If we decide to separate out the logic for ssm and secrets manager then we will need to add at least one more function to figure out if it is ssm or secrets manager and two for loops -- 1. fetch secrets manager and 2 fetch ssm params in CFN template. We can very well implement that but then complexity of the code will be increased when we can do the same task in the current one liner logic.